Automattic/wp-calypso
# Adding upgrade plugin screen information

Related to #76726, #77124

## Proposed Changes

* In this PR, we add screens to show upgrade plugin information if needed. In our new plugin flow, we show the plans page first and wait until a user clicks the migration button we then check for the plugin information. There's no point in checking it if a user is going to choose content-only import. In order to do so, we check it when:
    * User is on a plans page, and then clicks the `Upgrade and migrate` button
    * User already has a Business plan. We'll check it on page load.
    * When the upgrade screen shows up, it keeps pinging the API every five seconds, if user has established the connection, they can continue on the migration steps. 
* This PR also remove `sourceSite` object from being used if it's not needed.

![Screen Shot 2023-05-25 at 1 34 55 PM](https://github.com/Automattic/wp-calypso/assets/4074459/843a0a2e-2e8c-4be8-a3ca-04855ae2ec9a)

![Screen Shot 2023-05-25 at 1 49 39 PM](https://github.com/Automattic/wp-calypso/assets/4074459/c5778680-7a62-4f70-a6f1-2f19fbc9f63b)


## Testing Instructions



* Create a JN site, with and without WordPress.com user connected
* Navigate to `http://calypso.localhost:3000/setup/import-focused/importerWordpress?from=${source_site}&siteSlug=${target_site}&option=everything`
**Test case 1**
* Try with a simple site, you'll see the plans page first, click on `Upgrade and migrate` button
* It should come up with the upgrade plans screen if you don't have user account connected
* It should go to checkout page directly if you have Jetpack installed and user account connected
* Once it shows up the upgrade screen, it should go directly to the checkout page after you establish the connection.
**Test case 2**
* Try with an Atomic site, you'll see the migration ready screen if you have user account connected.
* It should show the upgrade screen if you don't have a user account connected
* Click on those action buttons on the upgrade screen and see if it guide you correctly

**Try with `Move to WordPress.com` scenario**

* Add this key value pair to your session storage in browser. key: `is_migrate_from_wp` value: `true` so we can identify it as a plugin flow
* Follow the steps above but install the Move to WordPress.com plugin with the current version, so if the upgrade plan screen shows up on a simple site
* Try another site with latest develop plugin version by creating it in JN site or Jetpack Beta, and see if you can go to migrate ready page without blocking.

## Pre-merge Checklist
